### PR TITLE
EQL: Backwards compatibility support for multi value fields update

### PR DESCRIPTION
**Pull request by astefan**
This is part of the work of introducing multi-value support in EQL. This covers the BWC checks.

### CODE REVIEW

File: x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java [hunk 1]
```diff
@@ -94,6 +101,91 @@ public void testSequencesWithRequestToUpgradedNodes() throws Exception {
...
+        Request bulkRequst = new Request("POST", index + "/_bulk?refresh");
+        bulkRequst.setJsonEntity(bulkEntries);
+        assertOK(client().performRequest(bulkRequst));
+
+        // build a set of functions names to check if all functions are tested with multi-value fields
+        final Set<String> availableFunctions = new EqlFunctionRegistry().listFunctions()
+            .stream()
+            .map(FunctionDefinition::name)
+            .collect(Collectors.toSet());
+        // each function has a query and a query results associated to it
```
Comment (bpintea) [inline x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java:120]: nit:
```suggestion
        // each function has a query and query results associated to it
```

File: x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java [hunk 2]
```diff
@@ -222,6 +314,23 @@ private String randomEvent() {
         return expectedResponse;
     }
 
+    private void assertMultiValueFunctionQuery(Set<String> availableFunctions, Set<String> testedFunctions, Request request,
+        RestClient client, String functionName, String query, int[] ids) throws IOException {
+        // trivial checks on the validity of functions being used
+        assertTrue("Function [" + functionName + "] not found in the list of functions " + availableFunctions,
+            availableFunctions.contains(functionName));
+        assertTrue("Function [" + functionName + "] not found in query [" + query + "].",
+            query.contains(functionName + "(") || query.contains(functionName + "~("));
```
Comment (bpintea) [inline x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java:323]: FMI: are these needed, given the last check of `testMultiValueFields()`? (reminded me of this anti-["defensive programming"](https://github.com/elastic/elasticsearch/pull/66022#discussion_r557676794) recommendation)

File: x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java [hunk 3]
```diff
@@ -222,6 +314,23 @@ private String randomEvent() {
...
+
+        List<Object> eventIds = new ArrayList<>();
+        for(int id : ids) {
```
Comment (bpintea) [inline x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java:326]: spacing. (I thought the formatting check would catch this.)

**Review by bpintea — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java [hunk 4]
```diff
@@ -222,6 +314,23 @@ private String randomEvent() {
         return expectedResponse;
     }
 
+    private void assertMultiValueFunctionQuery(Set<String> availableFunctions, Set<String> testedFunctions, Request request,
+        RestClient client, String functionName, String query, int[] ids) throws IOException {
+        // trivial checks on the validity of functions being used
+        assertTrue("Function [" + functionName + "] not found in the list of functions " + availableFunctions,
+            availableFunctions.contains(functionName));
```
Comment (Luegg) [inline x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java:321]: Isn't this check redundant? I would expect the query to fail anyway if the function is not in the registry.

**Review by Luegg — APPROVED** [review]
Lgtm 👍

File: x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java (hunk 2 above)
Comment (astefan) [reply x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java:323]: Fair point. I'll remove them.

File: x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java (hunk 4 above)
Comment (astefan) [reply x-pack/plugin/eql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/eql/qa/mixed_node/EqlSearchIT.java:321]: Removed it. Thanks.

**Pull request merged**